src/main/java/com/nerodesk/om/aws/AwsDoc.java

```diff
@@ -120,6 +120,11 @@ public void read(@NotNull final OutputStream output) throws IOException {
     @Override
     public void write(final InputStream input, final long size)
         throws IOException {
+        if (size == 0) {
```

Comment:
@dmzaytsev We already have may examples of such decorations in our code. For example, SmallDoc.
Or in XHT project. This is what Yegor expected to see. This is how Open/Closed principle works (http://en.wikipedia.org/wiki/Open/closed_principle)
As far as the check is completely the same in both places we should use decoration.

For null parameter we use annotation, so it's not the code duplication.